cli/internal/cmd/verify.go

```diff
@@ -447,6 +463,156 @@ func (f *attestationDocFormatterImpl) parseSNPReport(b *strings.Builder, reportB
 	return nil
 }
 
+func parseMAAToken(ctx context.Context, b *strings.Builder, rawToken, attestationServiceURL string) error {
+	var claims maaTokenClaims
+	_, err := jwt.ParseWithClaims(rawToken, &claims, keyFromJKUFunc(ctx, attestationServiceURL), jwt.WithIssuedAt())
+	if err != nil {
+		return err
+	}
+
+	out, err := json.MarshalIndent(claims, "\t\t", "  ")
+	if err != nil {
+		return err
+	}
+
+	b.WriteString("\tMicrosoft Azure Attestation Token:\n\t")
+	b.WriteString(string(out))
+	return nil
+}
+
+// keyFromJKUFunc returns a function that gets the JSON Web Key URI from the token
+// and fetches the key from that URI. The keys are then parsed, and the key with
+// the kid that matches the token header is returned.
+func keyFromJKUFunc(ctx context.Context, webKeysURL string) func(token *jwt.Token) (any, error) {
+	return func(token *jwt.Token) (any, error) {
+		if token.Header["alg"] != "RS256" {
+			return nil, fmt.Errorf("invalid signing algorithm: %s", token.Header["alg"])
+		}
+		kid, ok := token.Header["kid"].(string)
+		if !ok {
+			return nil, fmt.Errorf("invalid kid: %v", token.Header["kid"])
+		}
+		jku, ok := token.Header["jku"].(string)
+		if !ok {
+			return nil, fmt.Errorf("invalid jku: %v", token.Header["jku"])
+		}
+		if jku != webKeysURL+"/certs" {
+			return nil, fmt.Errorf("jku from token (%s) does not match configured attestation service (%s)", jku, webKeysURL)
+		}
+
+		ctx, cancel := context.WithCancel(ctx)
+		defer cancel()
+		keySetBytes, err := httpGet(ctx, jku)
```

Comment:
What's the effect of deriving the new context here instead of using the existing one?